### PR TITLE
Climbing and trap

### PR DESCRIPTION
**Pull request by ShayKalisher1234**
Climbing and trap

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/SubsystemTrap.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import com.ctre.phoenix.motorcontrol.can.TalonSRX;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
+import com.ctre.phoenix6.hardware.TalonFX;
+import static frc.robot.Constants.trapConstants.*;
+
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+public class SubsystemTrap extends SubsystemBase {
+  private WPI_TalonSRX m_trapMotor;
+  double speed;
```
Comment (ori-coval) [inline src/main/java/frc/robot/subsystems/SubsystemTrap.java]: no need for `double speed;`

File: src/main/java/frc/robot/subsystems/ClimbingSubsystem.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  {
+   if (instance == null)
+    {
+      instance = new ClimbingSubsystem();
+    }
+    return instance;
+  }
+  
+
+  public ClimbingSubsystem()
```
Comment (ori-coval) [inline src/main/java/frc/robot/subsystems/ClimbingSubsystem.java]: constructor should be private

File: src/main/java/frc/robot/commands/ClimbingCommands/SetSpeedRightMotorClimbing.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+package frc.robot.commands.ClimbingCommands;
+
+import edu.wpi.first.wpilibj2.command.InstantCommand;
+import frc.robot.subsystems.ClimbingSubsystem;
+
+// NOTE:  Consider using this command inline, rather than writing a subclass.  For more
+// information, see:
+// https://docs.wpilib.org/en/stable/docs/software/commandbased/convenience-features.html
+public class SetSpeedRightMotorClimbing extends InstantCommand {
+  private final ClimbingSubsystem rightMotorClimbing = ClimbingSubsystem.getInstance();
```
Comment (ori-coval) [inline src/main/java/frc/robot/commands/ClimbingCommands/SetSpeedRightMotorClimbing.java]: change `rightMotorClimbing` to `climbingSubsystem`

File: src/main/java/frc/robot/commands/ClimbingCommands/SetSpeedLeftMotorClimbing.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+package frc.robot.commands.ClimbingCommands;
+
+import edu.wpi.first.wpilibj2.command.InstantCommand;
+import frc.robot.subsystems.ClimbingSubsystem;
+
+// NOTE:  Consider using this command inline, rather than writing a subclass.  For more
+// information, see:
+// https://docs.wpilib.org/en/stable/docs/software/commandbased/convenience-features.html
+public class SetSpeedLeftMotorClimbing extends InstantCommand {
+  private final ClimbingSubsystem leftMotorClimbing = ClimbingSubsystem.getInstance();
```
Comment (ori-coval) [inline src/main/java/frc/robot/commands/ClimbingCommands/SetSpeedLeftMotorClimbing.java]: change `leftMotorClimbing`to `climbingSubsystem`

File: src/main/java/frc/robot/commands/ClimbingCommands/ClimbingMotorsSetSpeed.java
Comment (ori-coval) [inline src/main/java/frc/robot/commands/ClimbingCommands/ClimbingMotorsSetSpeed.java:1]: use instant command

File: src/main/java/frc/robot/Constants.java [hunk 5]
```diff
@@ -222,5 +222,13 @@ public static class Vision {
     public static final Matrix<N3, N1> kLeftMultiTagStdDevs = VecBuilder.fill(0.5, 0.5, 1);
 
     public static final Pose2d target = new Pose2d(1, 1, new Rotation2d(Units.degreesToRadians(0)));
+  }
+  public static class trapConstants {
+    public static final int TRAP_MOTOR_ID = 0;
+    public static final int M_CLIMBINGRIGHT_MOTOR_ID = 1;
+    public static final int M_CLIMBINGLEFT_MOTOR_ID = 2;
```
Comment (ori-coval) [inline src/main/java/frc/robot/Constants.java]: climbing constants should be in a different constants class

**Review by ori-coval — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/ClimbingSubsystem.java (hunk 2 above)
Comment (ShayKalisher1234) [reply src/main/java/frc/robot/subsystems/ClimbingSubsystem.java]: fixed

File: src/main/java/frc/robot/commands/ClimbingCommands/SetSpeedRightMotorClimbing.java (hunk 3 above)
Comment (ShayKalisher1234) [reply src/main/java/frc/robot/commands/ClimbingCommands/SetSpeedRightMotorClimbing.java]: fixed

File: src/main/java/frc/robot/commands/ClimbingCommands/SetSpeedLeftMotorClimbing.java (hunk 4 above)
Comment (ShayKalisher1234) [reply src/main/java/frc/robot/commands/ClimbingCommands/SetSpeedLeftMotorClimbing.java]: fixed

File: src/main/java/frc/robot/commands/ClimbingCommands/ClimbingMotorsSetSpeed.java
Comment (ShayKalisher1234) [reply src/main/java/frc/robot/commands/ClimbingCommands/ClimbingMotorsSetSpeed.java:1]: fixed

File: src/main/java/frc/robot/Constants.java (hunk 5 above)
Comment (ShayKalisher1234) [reply src/main/java/frc/robot/Constants.java]: fixed

**Review by ori-coval — APPROVED** [review]

**Pull request merged**